pkg/kube/cluster/cluster.go

```diff
@@ -57,6 +57,9 @@ func ShortNameN(clusterName string, maxLen int) string {
 	if len(clusterName) > maxLen {
 		shortClusterName = clusterName[0:maxLen]
 	}
+
+	shortClusterName = strings.ReplaceAll(shortClusterName, ".", "-")
```

Comment:
If this is arbitrary user input, it might be better to use one of the existing functions in [kube/names.go](https://github.com/jenkins-x/jx/blob/b618ee1546ff230c4b3c5e9c0c8f9c512a1b8dea/pkg/kube/names.go).

Reply:
Updated patch to use the code from kube/names.go